video_xblock/tests/mocks.py

```diff
@@ -48,29 +48,42 @@ class BaseMock(Mock):
     """
     Base custom mock class.
     """
-    event_types = []
-    event_results = []
+    # `outcomes` should be in the format of dict().items() to keep the order of items.
+    # First argument: result name, second argument - dictionary containing result data.
+    # Example: (("key1", {}), ("key2", {}), ...)
+    outcomes = ()
     to_return = []
 
     def __init__(self, **kwargs):
         super(BaseMock, self).__init__()
         if 'mock_magic' in kwargs:
             self.mock = kwargs['mock_magic']
 
+    @property
+    def ordered_results(self):
+        """
+        Transforms `outcomes` to dict.
+        """
+        return dict(self.outcomes)
```

Comment:
If you want to ensure order use `collections.OrderedDict`. Simple `dict` will unlikely do that.